tests/components/tasmota/test_sensor.py

```diff
@@ -120,6 +255,76 @@ async def test_attributes(hass, mqtt_mock, setup_tasmota):
     assert state.attributes.get("unit_of_measurement") == "ppm"
 
 
+async def test_nested_sensor_attributes(hass, mqtt_mock, setup_tasmota):
+    """Test correct attributes for sensors."""
+    config = copy.deepcopy(DEFAULT_CONFIG)
+    sensor_config = copy.deepcopy(NESTED_SENSOR_CONFIG)
+    mac = config["mac"]
+
+    async_fire_mqtt_message(
+        hass,
+        f"{DEFAULT_PREFIX}/{mac}/config",
+        json.dumps(config),
+    )
+    await hass.async_block_till_done()
+    async_fire_mqtt_message(
+        hass,
+        f"{DEFAULT_PREFIX}/{mac}/sensors",
+        json.dumps(sensor_config),
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get("sensor.tx23_speed_act")
+    assert state.attributes.get("device_class") is None
+    assert state.attributes.get("friendly_name") == "TX23 Speed Act"
+    assert state.attributes.get("icon") is None
+    assert state.attributes.get("unit_of_measurement") == "km/h"
+
+    state = hass.states.get("sensor.tx23_dir_avg")
+    assert state.attributes.get("device_class") is None
+    assert state.attributes.get("friendly_name") == "TX23 Dir Avg"
+    assert state.attributes.get("icon") is None
+    assert state.attributes.get("unit_of_measurement") == " "
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)